TACC-Cloud/hazmapper
# task/WG-249: Mirror map name and file name initially

## Overview: ##
In an effort to match angular hazmapper's mirroring feature (initially typing in the map name also populated the custom file name field), I mimicked that feature in react hazmapper.  
## PR Status: ##


## Related Jira tickets: ##

* [WG-249](https://tacc-main.atlassian.net/browse/WG-249)

## Summary of Changes: ##

- Added useEffect hook to watch changes in the map name and system file fields (name and custom file name)
- Now if the user types in the Name field, the Custom File Name will mirror the value, initially. However, if the user wants to make the Custom File Name different than the Map name, they can manually edit the Custom File Name field to do so.
- The Name field should be allowed to have spaces in the map name--however the Custom File Name field is not allowed to have spaces--so I added code to replace spaces to underscores in the Custom File Name field when mirroring. 

## Testing Steps: ##
Note: This needs to be run with local geoapi backend now
1. npm ci
2. Go to localhost:4200
3. Click on the Create Map button
4. Start typing in the Name field and watch that the Custom File Name mirrors the value
5. Now go to the Custom File Name field and edit it. Make sure the Name field doesn't change

## UI Photos:

## Notes: ##
